spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -880,7 +880,19 @@ private RuntimeException doInvokeRecordListener(final ConsumerRecord<K, V> recor
 						this.listener.onMessage(record);
 						break;
 				}
-				if (!this.isAnyManualAck && !this.autoCommit) {
+				if (this.isRecordAck) {
```

Comment:
Hm. Shouldn't we apply the same immediate logic for the `this.isBatchAck` mode in the `doInvokeBatchListener()`?

Shouldn't we skip then both of them in the `processCommits()` afterwards?

Reply:
Not necessary since we call `processCommits` anyway before the next poll, which will commit the batch acks.

I suppose we could now suppress the call to processCommits() if `isRecordAck` is true; will push.